ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/ClientOptionsImpl.java

```diff
@@ -86,6 +89,8 @@ private ClientOptionsImpl(
     this.executeQueryMaxResultRows = executeQueryMaxResultRows;
     this.http2MultiplexingLimit = http2MultiplexingLimit;
     this.requestHeaders = requestHeaders;
+    this.idpConfig = idpConfig;
+    this.authType = authType;
```

Comment:
do we want the user to pass `authType` given this will be set using setBasicAuth/setIdpConfig?

Reply:
User will never pass authType. It will be set by the code.

Reply:
User creates clientOptions and not clientOptionsImpl.

Reply:
ClientOption is an interface. The instantiation of this class does not require authType to be set in the constructor. It will be overridden by the `setIdpConfig` or `setBasicAuthCredentials` method call. And if these methods are not called, it should be in `None` state. Constructor taking `authType` allows it to set `Basic` or `Bearer` despite setter of one of these has not been called.